src/html-error.c

```diff
@@ -172,21 +172,37 @@ int send_http_error_message (struct conn_s *connptr)
             "<p><em>Generated by %s.</em></p>\n" "</body>\n"
             "</html>\n";
 
-        const char p_auth_str[] =
-            "Proxy-Authenticate: Basic realm=\""
-            PACKAGE_NAME "\"\r\n";
-
-        const char w_auth_str[] =
-            "WWW-Authenticate: Basic realm=\""
-            PACKAGE_NAME "\"\r\n";
-
 	/* according to rfc7235, the 407 error must be accompanied by
            a Proxy-Authenticate header field. */
-        const char *add = connptr->error_number == 407 ? p_auth_str :
-                          (connptr->error_number == 401 ? w_auth_str : "");
+        const char *auth_str_type = 
+                connptr->error_number == 407 ? "Proxy-Authenticate" :
+                (connptr->error_number == 401 ? "WWW-Authenticate" : "");
+
+        const char auth_str_tpl[] = "%s: Basic realm=\"%s\"\r\n";
+        char* auth_str_add = NULL;
+
+        if (auth_str_type[0] != 0) {
+                int auth_str_size = snprintf(NULL, 0, auth_str_tpl, 
+                                        auth_str_type, config->basicauth_realm) + 1;
+                if (auth_str_size > 0) {
+                        auth_str_add = safemalloc(auth_str_size);
+                        if (auth_str_add != NULL) {
+                                snprintf(auth_str_add, auth_str_size, auth_str_tpl,
+                                        auth_str_type, config->basicauth_realm);
+                        } else {
```

Comment:
i dont think we need any error handling here and below. if any of the things here fail, it's better to send the error code without authenticate string, which happens automatically because auth_str_add stays 0, than to terminate and send *nothing* at all.

Reply:
@rofl0r 
meanwhile i think i understand what you mean.

Then this way

```
        char* auth_str_add = NULL;

        if (auth_str_type[0] != 0) {
                int auth_str_size = snprintf(NULL, 0, auth_str_tpl, 
                                        auth_str_type, config->basicauth_realm) + 1;
                if (auth_str_size > 0) {
                        auth_str_add = safemalloc(auth_str_size);
                        if (auth_str_add != NULL) {
                                snprintf(auth_str_add, auth_str_size, auth_str_tpl,
                                        auth_str_type, config->basicauth_realm);
                        }
                }
        }
```